web/modules/custom/joinup_invite/src/Form/InviteToGroupForm.php

```diff
@@ -98,33 +131,168 @@ public function buildForm(array $form, FormStateInterface $form_state, RdfInterf
     return parent::build($form, $form_state);
   }
 
+  /**
+   * {@inheritdoc}
+   *
+   * Ensure that users are not already invited or members of the group.
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    if ($this->rdfEntity->bundle() !== 'collection') {
+      return;
+    }
+
+    $users = $this->getUserList($form_state);
+    $invalid_users = [];
+
+    foreach ($users as $user) {
+      $invitations = $this->entityTypeManager->getStorage('invitation')->loadByProperties([
+        'entity_type' => $this->rdfEntity->getEntityTypeId(),
+        'entity_id' => $this->rdfEntity->id(),
+        'recipient_id' => $user->id(),
+        'bundle' => 'group',
+      ]);
+      $invitation = reset($invitations);
+
+      $membership_states = [
+        OgMembershipInterface::STATE_ACTIVE,
+        OgMembershipInterface::STATE_BLOCKED,
+        OgMembershipInterface::STATE_PENDING,
+      ];
+      $membership = $this->ogMembershipManager->getMembership($this->rdfEntity, $user, $membership_states);
+
+      if (!empty($invitation) || !empty($membership)) {
```

Comment:
I thought that this would end up in a remark. I was struggling on that myself before I decided.
The issue with that is that while an invitation exists, a membership exists too. They are connected. So the first check (for memberships) will already bring positive results if the users is invited (so the second check is redundant).
The main reason though was a logical one. In my experience (not big but still), we are looking it in from a technical perspective and we want to know what is going on. However, for the end user, this is not always the case. I thought that we do not need to tell the end user whether a membership already exists or whether an invitation already exists because anyway the user will have limited options. Also, most of the cases I can think that the user will be invited while he is already invited is that either there was a complaint for the membership, or the facilitator did not know it.
In the first case, the user simply needs to delete the membership to revoke the invitations (until the invitation dashboard is created).
In the second case, the user needs to inform the user to take action.
In any case, the facilitator will be able to also simply go and approve his membership.

I mean, this can be a matter of discussion, but since we have no clear specification for this and since all available actions (delete membership, re invite, approve membership, respond to user) can be taken from the user himself, I think that it is simpler to show a simpler message.

Reply:
When I said "First check for memberships" I meant "First check for **active** memberships". But I disagree with you. I think it's a valuable information for the user to understand which account is already member and which is not yet active member but has been already invited. Keep in mind that the current user is a facilitator not regular user. He need to receive quality info from the site. Yes, I'm keeping my point here.